drivers/hmis/app/graphql/mutations/delete_assessment.rb

```diff
@@ -14,10 +14,11 @@ def resolve(id:)
       record = Hmis::Hud::CustomAssessment.viewable_by(current_user).find_by(id: id)
       raise HmisErrors::ApiError, 'Record not found' unless record.present?
 
-      # FIXME(#185552893): this mutation should delete all the related records linked to the FormProcessor
-      record.transaction do
+      record.with_lock do
         is_wip = record.in_progress?
 
+        raise HmisErrors::ApiError, 'Cannot reopen an enrollment with an entry date before other enrollments' if record.deletion_would_cause_conflicting_enrollments?
```

Comment:
Great! Thanks for adding this.
However - raising ApiError gives the user a "server error" message. The message here would only be shown to the user in development.
For user-decipherable errors, we should return an `HmisErrors::Error` class like `{ errors: [error] }`.